packages/grid/x-data-grid-pro/src/DataGridPro/DataGridPro.tsx

```diff
@@ -11,30 +11,18 @@ import {
   GridContextProvider,
 } from '@mui/x-data-grid';
 import { useDataGridProComponent } from './useDataGridProComponent';
-import { Watermark } from '../components/Watermark';
 import { DataGridProProps } from '../models';
 import { useDataGridProProps } from './useDataGridProProps';
 import { DataGridProVirtualScroller } from '../components/DataGridProVirtualScroller';
 import { DataGridProColumnHeaders } from '../components/DataGridProColumnHeaders';
 
-// This is the package release date. Each package version should update this const
-// automatically when a new version is published on npm.
-let RELEASE_INFO = '__RELEASE_INFO__';
-
-// eslint-disable-next-line no-useless-concat
-if (process.env.NODE_ENV !== 'production' && RELEASE_INFO === '__RELEASE' + '_INFO__') {
-  // eslint-disable-next-line no-underscore-dangle
-  RELEASE_INFO = ponyfillGlobal.__MUI_RELEASE_INFO__;
-}
-
-LicenseInfo.setReleaseInfo(RELEASE_INFO);
-
 const DataGridProRaw = React.forwardRef<HTMLDivElement, DataGridProProps>(function DataGridPro(
   inProps,
   ref,
 ) {
   const props = useDataGridProProps(inProps);
   const apiRef = useDataGridProComponent(props.apiRef, props);
+  useLicenseVerifier();
```

Comment:
What's the motivation for not having Watermark do it?

A related matter, how do we display a watermark with the date range picker? maybe on the background of the page? 🤔

Reply:
When doing the headless version, we need to check the license even if no watermark is called

Reply:
![image](https://user-images.githubusercontent.com/3309670/158552694-0438cc76-018a-43be-8f50-eabc05e82971.png)

Reply:
For the check, I memoize it so calling `useLicenseVerifier` a lot of time is not a big problem I think.
For the date pickers I call it in `DateRangePicker`, in `MobileDateRangePicker`, in `DesktopDateRangePicker`, in `StaticDateRangePicker` etc... basically in each pro component exported. 
That removes the struggle to think whether or not the check is done in every scenario. The hook is called several time, the 1st time it warns, then it just returns the memoized status and does nothing more.

Reply:
Ok, in the future, I think that we could consider rendering a watermark portal under the `<body>`, position fixed, similar to an Intercom widget or a https://www.netlify.com/products/deploy-previews widget.